### PR TITLE
Only paint the dirty rect of non-tiled blobs.

### PR DESCRIPTION
**Pull request by nical**
Unless we settle with a small tile size we should follow up with the same optimization for tiled blobs.
This is built on top of #2941.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2948)

### CODE REVIEW

**Comment by nical** [comment]
try push with reftests green https://treeherder.mozilla.org/#/jobs?repo=try&revision=3ea9ba2961163832ab9b4279d0b97d0d46dd3c90

File: examples/blob.rs [hunk 1]
```diff
@@ -97,7 +98,10 @@ fn render_blob(
 
     Ok(api::RasterizedBlobImage {
         data: Arc::new(texels),
-        size: descriptor.size,
+        rasterized_rect: DeviceUintRect {
```
Comment (kvark) [inline examples/blob.rs]: nit: could use [from_size](https://docs.rs/euclid/0.19.0/euclid/struct.TypedRect.html#method.from_size)

**Pull request merged**